deblur/workflow.py

```diff
@@ -207,6 +208,65 @@ def build_index_sortmerna(ref_fp, working_dir):
     return all_db
 
 
+def remove_artifacts_from_biom_table(table_filename,
+                                     fasta_filename,
+                                     ref_fp,
+                                     biom_table_dir,
+                                     ref_db_fp,
+                                     threads=1,
+                                     verbose=False,
+                                     sim_thresh=None,
+                                     coverage_thresh=None):
+    """Remove artifacts from a biom table using SortMeRNA
+
+    Parameters
+    ----------
+    table : biom.Table or str
+        the biom table to remove artifacts from or name of the biom table file
+    fasta_filename : str
+        the fasta file containing all the sequences of the biom table
+    """
+    logger = logging.getLogger(__name__)
+    logger.info('getting 16s sequences from the biom table')
+
+    # remove artifacts from the fasta file. output is in clean_fp fasta file
+    clean_fp = remove_artifacts_seqs(fasta_filename, ref_fp,
+                                     working_dir=biom_table_dir,
+                                     ref_db_fp=ref_db_fp,
+                                     negate=False, threads=threads,
+                                     verbose=verbose,
+                                     sim_thresh=sim_thresh,
+                                     coverage_thresh=coverage_thresh)
+    logger.debug('removed artifacts from sequences input %s'
+                 ' to output %s' % (fasta_filename, clean_fp))
+
+    # read the clean fasta file
+    good_seqs = set()
+    for chead, cseq in sequence_generator(clean_fp):
+        if cseq not in good_seqs:
+            good_seqs.add(cseq)
+    logger.debug('loaded %d sequences from cleaned biom table'
+                 ' fasta file' % len(good_seqs))
+
+    logger.debug('loading biom table %s' % table_filename)
+    table = load_table(table_filename)
+
+    # filter and save the artifact biom table
+    artifact_table = table.filter(list(good_seqs),
```

Comment:
in general deblur does not do any sample filtering (the old version used to keep also samples with 0 reads... not sure about this version). Anyway, a first step in any downstream analysis should be removing samples with low reads/rarifying.
We can incorporate this into deblur, but i don't think it's the best approach (and where do we stop - filter samples with 0 reads only? or filter also samples with 1 read...)
what do you think?